crates/ruff_linter/src/rules/fastapi/rules/fastapi_non_annotated_dependency.rs

```diff
@@ -150,47 +143,127 @@ fn is_fastapi_dependency(checker: &Checker, expr: &ast::Expr) -> bool {
+/// ```
+///
+/// Gets fixed to
+///
+/// ```python
+/// def handler(some_path_param: Annotated[str, Path()]): pass
+/// ```
+///
+/// Causing it to lose its default value. That's fine in this example but causes a syntax error if
+/// `some_path_param` comes after another argument with a default. We only compute the information
+/// necessary to determine this while generating the fix, thus the need to return an updated
+/// `seen_default` here.
 fn create_diagnostic(
     checker: &mut Checker,
     parameter: &ast::ParameterWithDefault,
-    safe_to_update: bool,
-) {
+    mut seen_default: bool,
+) -> bool {
     let mut diagnostic = Diagnostic::new(
         FastApiNonAnnotatedDependency {
             py_version: checker.settings.target_version,
         },
         parameter.range,
     );
 
-    if safe_to_update {
-        if let (Some(annotation), Some(default)) =
-            (&parameter.parameter.annotation, &parameter.default)
-        {
-            diagnostic.try_set_fix(|| {
-                let module = if checker.settings.target_version >= PythonVersion::Py39 {
-                    "typing"
-                } else {
-                    "typing_extensions"
-                };
-                let (import_edit, binding) = checker.importer().get_or_import_symbol(
-                    &ImportRequest::import_from(module, "Annotated"),
-                    parameter.range.start(),
-                    checker.semantic(),
-                )?;
-                let content = format!(
-                    "{}: {}[{}, {}]",
-                    parameter.parameter.name.id,
-                    binding,
-                    checker.locator().slice(annotation.range()),
-                    checker.locator().slice(default.range())
-                );
-                let parameter_edit = Edit::range_replacement(content, parameter.range);
-                Ok(Fix::unsafe_edits(import_edit, [parameter_edit]))
+    if let (Some(annotation), Some(default)) = (&parameter.parameter.annotation, &parameter.default)
+    {
+        let mut try_generate_fix = || {
+            let module = if checker.settings.target_version >= PythonVersion::Py39 {
+                "typing"
+            } else {
+                "typing_extensions"
+            };
+            let (import_edit, binding) = checker.importer().get_or_import_symbol(
+                &ImportRequest::import_from(module, "Annotated"),
+                parameter.range.start(),
+                checker.semantic(),
+            )?;
+
+            // Refine the match from `is_fastapi_dependency` to exclude Depends
+            // and Security, which don't have the same argument structure. The
+            // others need to be converted from `q: str = Query("")` to `q:
+            // Annotated[str, Query()] = ""` for example, but Depends and
+            // Security need to stay like `Annotated[str, Depends(callable)]`
+            let is_dep = checker
```

Comment:
I find the name `is_dep` slightly confusing because it explicitly excludes `Depends`. How about: `is_route_param` `is_route`